scripts/sonar/sonar-controller.js

```diff
@@ -67,6 +84,31 @@ angular.module('app.sonar').controller('SonarCtrl', ['$scope', '$log', '$q', 'ap
             $log.info('Finished calculating metrics');
         }
 
+        function calculateFileComplexity(sonarData) {
+            var complexity = sum(sonarData, 'projectComplexity')
+            return complexity / sum(sonarData, 'numberOfFiles')
+        }
+        
+        function calculateMethodComplexity(sonarData){
+            var complexity = sum(sonarData, 'projectComplexity')
+            return complexity / sum(sonarData, 'numberOfMethods')
+        }
+        
+        function calculateLineCoverage(sonarData){
+            var testableLinesOfCode = sum(sonarData, 'projectTestableLines')
+            var untestedLinesOfCode = sum(sonarData, 'projectUntestableLines')      
+            
+            return (((testableLinesOfCode - untestedLinesOfCode) / testableLinesOfCode) * 100) || 0
+        }
+        
+        function calculateCompliance(sonarData, totalLines){
+            var totalWeightedIssues = sum(sonarData, 'weightedBlockingIssues') + sum(sonarData, 'weightedCriticalIssues')
+                                    + sum(sonarData, 'weightedMajorIssues') + sum(sonarData, 'weightedMinorIssues')
+                                    + sum(sonarData, 'weightedInfoIssues');
+                                    
+            return (100 - (totalWeightedIssues / totalLines) * 100)
```

Comment:
Your brackets are in the wrong place here, the SonarQube documentation says:

    (RCI) = 100 - (Weighted issues / Lines of code * 100)